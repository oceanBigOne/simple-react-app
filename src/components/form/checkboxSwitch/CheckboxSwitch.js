import React, {Component} from "react";
import style from "./CheckboxSwitch.module.scss";

class CheckboxSwitch extends Component {
    constructor(props) {
        super(props);
        this.state={
            focused:false
        }
    }

    sum(a,b){
        return a+b;
    }

    focus(){
        this.setState({ focused: true });
    }

    blur(){
        this.setState({ focused: false });
    }

    render() {
        return (
            <label className={(this.state.focused?style.focused:"")+" "+style.switch} >
                <input
                    type="checkbox"
                    id={this.props.id}
                    onFocus={() => this.setState({ focused: true })}
                    onBlur={() => this.setState({ focused: false })}
                    name={this.props.name}
                    tabIndex={this.props.tabIndex}
                    onClick={this.props.onClick}
                    className={this.props.className}
                />
                <span className={style.slider} />
            </label>
        );
    }
}

export default CheckboxSwitch;