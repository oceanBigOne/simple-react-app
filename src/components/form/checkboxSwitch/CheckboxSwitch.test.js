import React from 'react';
import CheckboxSwitch from "./CheckboxSwitch";
import renderer from 'react-test-renderer';


test('checkboxSwicth change on click', () => {
    const component = renderer.create(
        <CheckboxSwitch />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component.getInstance().focus();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

   // console.log(tree.children[0].props);
    // manually trigger the callback
   // tree.children[0].props.onFocus();
    // re-rendering
    //tree = component.toJSON();
    //expect(tree).toMatchSnapshot();
});

test('adds 1 + 2 to equal 3', () => {
    const component = renderer.create(
        <CheckboxSwitch />,
    );

    expect(component.getInstance().sum(1, 2)).toBe(3);
});