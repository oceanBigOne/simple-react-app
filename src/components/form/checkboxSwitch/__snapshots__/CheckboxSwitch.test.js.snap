// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkboxSwicth change on click 1`] = `
<label
  className=" switch"
>
  <input
    onBlur={[Function]}
    onFocus={[Function]}
    type="checkbox"
  />
  <span
    className="slider"
  />
</label>
`;

exports[`checkboxSwicth change on click 2`] = `
<label
  className="focused switch"
>
  <input
    onBlur={[Function]}
    onFocus={[Function]}
    type="checkbox"
  />
  <span
    className="slider"
  />
</label>
`;
